File: app/src/main/java/com/karthik/splash/DI/SplashAppComponent.java
package com.karthik.splash.DI;

import com.karthik.splash.Modules.FeedsHomeModule;
import com.karthik.splash.Modules.FeedsScreenModule;
import com.karthik.splash.Modules.HomeTabFeedsModule;
import com.karthik.splash.Modules.LikeTabFeedsModule;
import com.karthik.splash.Modules.SplashApiModule;
import com.karthik.splash.Modules.SplashAppModule;
import com.karthik.splash.Modules.SplashScreenModule;
import com.karthik.splash.SplashApp;

import javax.inject.Singleton;

import dagger.Component;

/**
 * Created by karthikrk on 12/11/17.
 */
@Singleton
@Component(modules = {SplashAppModule.class, SplashApiModule.class})
public interface SplashAppComponent {
    void inject(SplashApp splashApp);
    SplashScreenComponent plus(SplashScreenModule splashScreenModule);
    FeedsScreenComponent  plus(FeedsScreenModule feedsScreenModule);
    FeedsHomeComponent plus(FeedsHomeModule feedsHomeModule);
    HomeTabFeedsComponent plus(HomeTabFeedsModule homeTabFeedsModule);
    LikeTabFeedsComponent plus(LikeTabFeedsModule likeTabFeedsModule);
}

File: app/src/main/java/com/karthik/splash/root/SplashAppComponent.kt
package com.karthik.splash.root


import com.karthik.network.di.NetworkProvider
import com.karthik.splash.homescreen.di.HomeScreenComponent
import com.karthik.splash.homescreen.di.HomeScreenModule
import com.karthik.splash.photodetailscreen.di.PhotoDetailScreenComponent
import com.karthik.splash.photodetailscreen.di.PhotoDetailScreenModule
import com.karthik.splash.splashscreen.di.SplashScreenComponent
import com.karthik.splash.splashscreen.di.SplashScreenModule
import dagger.Component
import javax.inject.Singleton

@Singleton
@Component(modules = [SplashAppModule::class, NetworkModule::class, NetworkProvider::class])
interface SplashAppComponent {
    fun inject(app: SplashApp)
    fun plus(splashScreenModule: SplashScreenModule): SplashScreenComponent
    fun plus(homeScreenModule: HomeScreenModule): HomeScreenComponent
    fun plus(photoDetailModule: PhotoDetailScreenModule): PhotoDetailScreenComponent
}